### ISSUE TITLE
['[FAIL] ERROR:'] Nessus: Download Fails with Larger Scan

### ISSUE THREAD

**Issue by colsephiroth** [HASecuritySolutions/VulnWhisperer#63]
An authenticated scan of 1000 hosts generated a file around 500 MB in size when exported as a csv manually. This tool fails to download this scan, but after I removed that scan it succeeded at downloading a much smaller scan that targeted only 1 host.

Client: Archlinux 4.16.10-1-ARCH
VulnWhisperer: 1.5 and master
Python: 2.7.15

Server: Centos 7.5.1804
Nessus: Professional 7.1.0

(root@endor-vm) vulnwhisperer # python2 bin/vuln_whisperer -c configs/frameworks_example.ini -s nessus
[INFO] Creating directory /opt/vulnwhisperer/database
[INFO] Creating file /opt/vulnwhisperer/database/report_tracker.db
[INFO] Connected to database at /opt/vulnwhisperer/database/report_tracker.db
[INFO] Attempting to connect to nessus...
[SUCCESS] Connected to nessus on sherlock.strsoh.org:8834
[INFO] Gathering all scan data... this may take a while...
[INFO] Identified 2 scans to be processed
Download for file id 1720763221.
..............................
...............
[FAIL] ERROR: 

(root@endor-vm) vulnwhisperer # python2 bin/vuln_whisperer -c configs/frameworks_example.ini -s nessus
[INFO] Connected to database at /opt/vulnwhisperer/database/report_tracker.db
[INFO] Attempting to connect to nessus...
[SUCCESS] Connected to nessus on sherlock:8834
[INFO] Gathering all scan data... this may take a while...
[INFO] Identified 1 scans to be processed
Download for file id 1331664840.
.
Processing 1/1 for scan: easton-mbx1
[INFO] 522 records written to easton-mbx1_55_56_1527889658.csv

**Comment by praveenchary** [comment]
Even i am experiencing the same problem.Any workaround you found to rule out this issue ?

Thanks in advance

Regards,
Praveen

**Comment by austin-taylor** [comment]
We need to improve logging a bit here. We read in each file to do a line count, but I'm not sure that's necessary. I'll try taking that out and see if it helps.

**Comment by pemontto** [comment]
I just tried this on a ~1GB file using the mocked interface. No issues so it's likely transient network issues that cause downloads to fail.

```bash
$> ls -lh data/nessus/GET_scans_exports_164_download
-rw-r--r--  1 dev  staff   1.2G  8 Apr 10:17 data/nessus/GET_scans_exports_164_download
```


```
DEBUG:NessusAPI:request:Requesting to url https://nessus:443/scans/exports/164/download
DEBUG:urllib3.connectionpool:_new_conn:Starting new HTTPS connection (1): nessus
DEBUG:urllib3.connectionpool:_make_request:https://nessus:443 "GET /scans/exports/164/download HTTP/1.1" 200 1242154788
DEBUG:NessusAPI:request:Returning data.content
INFO:vulnWhispererNessus:whisper_nessus:Processing 2/2 for scan: VulnWhisperer - vulnweb.com ñ Nessus B
INFO:vulnWhispererNessus:whisper_nessus:1000222 records written to VulnWhisperer_-_vulnweb.com_ñ_Nessus_B_164_2_1554002688.csv
INFO:vulnWhispererNessus:whisper_nessus:Scan aggregation complete! Connection to database closed.
```

**Comment by reactor7171** [comment]
I have been doing tests generating reports from Nessus, replicating hosts to simulate the same situation, and still I haven't managed to trigger the same situation with my Nessus instance (on a server in the network, simulating a real network environment).

Host used for replication: Metasploitable VM - 104 Vulnerabilities detected
Tests:
- 350 assets - 213 MB CSV
- 763 assets - 321 MB CSV
- 1271 assets - 534 MB CSV 

We have implemented sessions and download in chunks (#167) in order to make sure that we are handling the best on our side, but if we are talking about transient network issues, it is difficult from our side to find a solution to that issue.

We will consider closing this issue in its current status.

**Comment by reactor7171** [comment]
We had a new case if this bug with user @Blake, all of the conversation being in the VulnWhisperer Slack channel, and we managed to get to the root issue of this problem.

As VulnWhisperer is trying to download from Nessus all the data of the scan, this data is being stored in memory in a variable; at the moment that the size of the data is bigger than the available memory in the server, it crashes. @Blake was having a server with 8 cores and 16 GB of RAM, and managed to download scans up to 1.7GB, but there were still some that were crashing VulnWhisperer.
![image](https://user-images.githubusercontent.com/6662533/68749183-16c40a80-05fe-11ea-9f39-14b1c4cd9369.png)
After upgrading the servers RAM, VulnWhisperer managed to make it through the scans and download all the data successfully.
![image](https://user-images.githubusercontent.com/6662533/68749255-3824f680-05fe-11ea-9156-10bb1f098c67.png)

Thanks for all your help troubleshooting @Blake!
